addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -371,6 +378,24 @@ def _update_invoice_from_binary(self, filename, content, extension, invoice):
                 return self._import_fattura_pa(decoded_content, invoice)
         return super()._update_invoice_from_binary(filename, content, extension, invoice)
 
+    def _l10n_it_edi_search_tax(self, company, percentage, extra_domain=None):
+        extra_domain = extra_domain or []
+        conditions = [
+            ('company_id', '=', company.id),
+            ('amount', '=', percentage),
+            ('amount_type', '=', 'percent'),
+            ('type_tax_use', '=', 'purchase'),
+        ] + extra_domain
+        taxes = self.env['account.tax'].search(conditions).filtered(
+            lambda x: x.invoice_repartition_line_ids.mapped("factor_percent") == [100.0, 100.0])
```

Comment:
![image](https://user-images.githubusercontent.com/1665365/205276579-aaad312f-438d-46a1-a374-76b1e6715d27.png)
It's used 3 times here and there during import to look for existing taxes that match what's in the XML.
I can change the name to `_l10n_it_edi_import_search_tax` but... `_l10n_it_edi_search_tax` is enough imho.
We generally exclude ReverseCharge taxes because those are never imported.

Reply:
Done

Reply:
> why do you filter only the "invoice_repartition_line_ids" field?

La ricerca della tassa che stiamo importando avviene per aliquota (amount) tipo (purchase, acquisto) e per dominio aggiuntivo.  Tra i domini aggiuntivi ci sono il tipo di esenzione per tasse 0% (kind of exemption), il tipo ritenuta e il tipo di fondo pensione. La ricerca viene poi  *ulteriormente* filtrata per escludere le tasse Reverse Charge (che hanno linee di ripartizione base 100 -100 100  anziche' 100 100 ) che sicuramente non sono quelle da importare

The search for the tax we're importing happens by amount, by tax type and additional domain.
The additional domains can be the kind of exemption (for 0% taxes), the withholding type and the pension fund type. The search is then filtered *again* to exclude Reverse Charge taxes ( which have 100 -100 100 instead of 100 100) that sure aren't the ones we want to import.

![image](https://user-images.githubusercontent.com/1665365/205465582-5d52a030-84b9-48e3-b454-e289b69c05d7.png)
![image](https://user-images.githubusercontent.com/1665365/205465656-12b2518a-5b6c-4f5f-9ad9-2e7dc13356c0.png)

Reply:
@lordkrandel Ok, but if you have a refund invoice? would it be correct to also check on refund_repartition_line_ids?

Reply:
Ah! You mean when you import a refund! Yeah, right... 🤔🤔🤔🤔🤔🤔🤔🤔

Reply:
Well, on a second thought, it should be the same.

We want to exclude 100base -100 100 taxes.  The tax is the same for both invoices and refunds, right? So if I filter by invoice repartition line, I'm still excluding the -100 100  taxes.

If we want to be more precise in import, we will have to think about something more specific. But for now I think it's ok.
![image](https://user-images.githubusercontent.com/1665365/205971015-38a60763-add9-4d6e-b6f3-273260ef9d41.png)

Reply:
l10n_it_edi_search_tax_for_import  //  l10n_it_edi_search_tax_without_rc